js/app/courseHistoryStore.js

```diff
@@ -82,17 +82,12 @@ const CourseHistoryStore = new GObject.Class({
             tags_match_all: ['EknSetObject'],
             sort: Eknc.QueryObjectSort.SEQUENCE_NUMBER,
         });
-        Eknc.Engine.get_default().query(query, null, (engine, task) => {
-            let results;
-            try {
-                results = engine.query_finish(task);
-            } catch (error) {
-                logError(error);
-                return;
-            }
+        Eknc.Engine.get_default().query_promise(query).then((results) => {
```

Comment:
Style nit: when we were doing promises in the pantheon in node.js, we had settled on this style:
```js
Eknc.Engine.get_default().query_promise(query)
.then(results => {
    blah blah
})
.catch(error => {
    logError(error)
});
```
But since this is the first promise code we have, I'm open to setting new standards.

Also you may be able to do `.catch(logError)` which would be quite concise and clear IMO.